entries/pkid.go

```diff
@@ -95,3 +127,74 @@ func bulkDeletePkidEntry(entries []*lib.StateChangeEntry, db *bun.DB, operationT
 
 	return nil
 }
+
+func PkidBatchOperation(entries []*lib.StateChangeEntry, db *bun.DB, params *lib.DeSoParams) error {
+	// We check before we call this function that there is at least one operation type.
+	// We also ensure before this that all entries have the same operation type.
+	glog.Infof("PkidBatchOperation: Putting %d entries into the database", len(entries))
+	operationType := entries[0].OperationType
+	var err error
+	if operationType == lib.DbOperationTypeDelete {
+		err = bulkDeletePkid(entries, db, operationType)
+	} else {
+		err = bulkInsertPkid(entries, db, operationType, params)
+	}
+	if err != nil {
+		return errors.Wrapf(err, "entries.PostBatchOperation: Problem with operation type %v", operationType)
+	}
+	return nil
+}
+
+// bulkInsertPkid inserts a batch of PKIDs into the database.
+func bulkInsertPkid(entries []*lib.StateChangeEntry, db *bun.DB, operationType lib.StateSyncerOperationType, params *lib.DeSoParams) error {
+	// Track the unique entries we've inserted so we don't insert the same entry twice.
+	uniqueEntries := consumer.UniqueEntries(entries)
+	// NOTE: if we need to support parsing other indexes for PKIDs beyond LeaderSchedule,
```

Comment:
Just for the sake of completeness, lets do a prefix filter on the uniqueEntries to just the leader schedule prefix, and use that slice in the rest of the function.